gatsby-node.js

```diff
@@ -7,56 +7,122 @@ exports.sourceNodes = async (
   const { createNode } = actions
 
   // Get data from Confluence
-  const response = await search(pluginOptions)
-
-  const baseUrl = response._links.base
-  const results = response.results.filter(result => result.type === 'page')
+  const fetchURL = 'https://'+ pluginOptions.hostname +'/wiki/rest/api/content/search/?cql=('+ pluginOptions.cql +')&expand=body.view,metadata.labels,history,ancestors,children.attachment&limit='+ pluginOptions.limit
+  const response = await fetchRequest(fetchURL, pluginOptions.auth)
 
+  const baseUrl = 'https://'+ pluginOptions.hostname +'/wiki';
+  const pages = response.filter(result => result.type === 'page')
   // Parse into nodes and add to GraphQL schema
-  const nodes = results.map(pageResult =>
+  const nodes = pages.map(pageResult =>
     formatPageNode(createNodeHelperFunctions, pageResult, baseUrl)
   )
 
   nodes.forEach(node => {
     // Create node
     createNode(node)
   })
+  
+
 }
 
-const search = async ({ hostname, auth, cql, limit = 10 }) => {
-  return await fetch(
-    `https://${hostname}/wiki/rest/api/content/search/?cql=(${cql})&expand=body.view,history,ancestors&limit=${limit}`,
+var dataOut = [];
+//based on https://stackoverflow.com/questions/62337587/return-paginated-output-recursively-with-fetch-api
+async function fetchRequest(url, auth) {
+  try {
+    // Fetch request and parse as JSON
+    const response = await fetch(url,
     {
       headers: {
         Authorization: auth,
         Accept: 'application/json',
       },
+    });
+    let data = await response.json();
+    dataOut = dataOut.concat(data.results);
+
+    // Extract the url of the response's "next"
+    let next_page;
+    if ( data.hasOwnProperty("_links") ) {
+      if ( data._links.hasOwnProperty("next") ) {
+        next_page = data._links.base + data._links.next;
+      }
     }
-  )
-    .then(x => x.json())
-    .catch(e => {
-      console.error('Unable to retrieve data from Confluence:', e)
-      process.exit(1)
-    })
+
+    // If another page exists, merge its output into the array recursively
+    console.log('Fetch next_page: ' + next_page)
+    if (next_page) {
+      dataOut.concat( await fetchRequest(next_page, auth) );
+    }
+    return dataOut;
+  } catch (err) {
+    return console.error(err);
+  }
+}
+
+function updateLinksInHTML(html) {
+
+  const regexLinks = /href\s*=\s*(['"])(https?:\/\/eugem.+?|\/wiki\/.+?)\1/gm;
```

Comment:
What is the eugem section for? Given your username, I'm assuming it's something specific to you?

Reply:
Apologies, I saw this only now. Yes, it is something specific to our project - the "eugem" string should not be hardcoded but passed as a parameter. (it is the string before "atlassian.net")